pkg/apis/resource/validation/validation.go

```diff
@@ -505,7 +505,19 @@ func validateResourceSliceSpec(spec, oldSpec *resource.ResourceSliceSpec, fldPat
 		allErrs = append(allErrs, field.Required(fldPath, "exactly one of `nodeName`, `nodeSelector`, or `allNodes` is required"))
 	case 1:
 	default:
-		allErrs = append(allErrs, field.Invalid(fldPath, nil, "exactly one of `nodeName`, `nodeSelector`, or `allNodes` is required"))
+		setFields := make([]string, 0)
+		if spec.NodeName != "" {
+			setFields = append(setFields, "`nodeName`")
+		}
+		if spec.NodeSelector != nil {
+			setFields = append(setFields, "`nodeSelector`")
+		}
+		if spec.AllNodes {
+			setFields = append(setFields, "`allNodes`")
+		}
+		allErrs = append(allErrs, field.Invalid(fldPath, nil,
+			fmt.Sprintf("exactly one of `nodeName`, `nodeSelector`, or `allNodes` is required, but multiple fields are set: %s", strings.Join(setFields, ", ")),
+		))
```

Comment:
I believe I had implemented something like that myself and then @thockin suggested to simplify it. My implementation created the set while checking fields, in contrast to this one here where the details get collected only when needed.

I'll defer to @thockin.